### PR TITLE
ICU-21109 Add support for locale-dependent minimum grouping digits to DecimalFormat

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21109
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [x] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/i18n/number_grouping.cpp [hunk 1]
```diff
@@ -64,6 +64,13 @@ Grouper Grouper::forProperties(const DecimalFormatProperties& properties) {
 }
 
 void Grouper::setLocaleData(const impl::ParsedPatternInfo &patternInfo, const Locale& locale) {
+    if (fMinGrouping == -2) {
```
Comment (FrankYFTang) [inline icu4c/source/i18n/number_grouping.cpp:67]: I have hard time to understand how the code jump from DecimalFormat::setMinimumGroupingDigits to here. Could you explain the relationship between  DecimalFormat and Grouper? 
Is there a design doc I can see to understand the internal relationship between these classes/objects?

Comment (sffc) [reply icu4c/source/i18n/number_grouping.cpp:67]: Grouper is for NumberFormatter, but also indirectly for DecimalFormat.  The DecimalFormat properties gets turned into NumberFormatter properties via number_mapper.cpp.

Comment (sffc) [reply icu4c/source/i18n/number_grouping.cpp:67]: Note that -2 and -3 are already supported internally in NumberFormatter.  I'm just exposing them to DecimalFormat.

I had to move these 7 lines of code to the top of the function in order for them to get evaluated even when fGrouping1 and fGrouping2 are set, as is the case in the DecimalFormat code path.

**Comment by FrankYFTang** [comment]
This is obsoleted by https://github.com/unicode-org/icu/pull/1152

**Pull request closed without merging**